ldap/servers/slapd/pagedresults.c

```diff
@@ -350,7 +354,7 @@ pagedresults_free_one(Connection *conn, Operation *op, int index)
  * Used for abandoning - pageresult_lock_get_addr(conn) is already locked in do_abandone.
  */
 int
-pagedresults_free_one_msgid_nolock(Connection *conn, ber_int_t msgid)
+pagedresults_free_one_msgid(Connection *conn, ber_int_t msgid, pthread_mutex_t *mutex)
```

Comment:
Why not retrieving the mutex in the function itself with pageresult_lock_get_addr(conn) ?

Reply:
Why recomputing the hash in the function when the caller already have computed it ( as in opshared.c case ) ?

Reply:
exact